src/main/java/org/spongepowered/common/item/inventory/EmptyInventoryImpl.java

```diff
@@ -225,6 +225,21 @@ public void setMaxStackSize(int size) {
     }
 
     @Override
+    public Inventory intersect(Inventory inventory) {
+        return this; // Shortcut
+    }
+
+    @Override
+    public Inventory union(Inventory inventory) {
+        return inventory; // Shortcut
+    }
+
+    @Override
+    public boolean containsInventory(Inventory inventory) {
+        return this == inventory;
```

Comment:
Well i'm not sure.
If this were strictly a set operation it should still return true for other empty inventories.
But as containsInventory checks for the actual structure being inside the other inventory I think this is correct.

@Mumfrey ?